spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -244,6 +245,11 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
       val snapshot = table.update()
       deltaLog.protocolWrite(snapshot.protocol)
 
+      if (table.catalogTable.exists(_.tableType == CatalogTableType.MANAGED)
```

Comment:
Add a comment to explain this: Vacuum can break clones by removing files that the clone still references. For managed tables, we eventually want the catalog to track this dependency to avoid breaking clones, but for now we forbid running vacuum on managed tables.